foundationdb/src/tuple/mod.rs

```diff
@@ -35,6 +36,25 @@ pub enum Error {
     FromUtf8Error(FromUtf8Error),
 }
 
+/// Tracks the depth of a Tuple decoding chain
+#[derive(Copy, Clone)]
+pub struct TupleDepth(usize);
+
+impl TupleDepth {
+    pub(crate) fn new() -> Self {
```

Comment:
This is just brilliant. I love it!

Reply:
oops, that pub(crate) isn't necessary... let me remove that.